src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -181,14 +181,13 @@ def rsa_padding_supported(self, padding: AsymmetricPadding) -> bool:
         if isinstance(padding, PKCS1v15):
             return True
         elif isinstance(padding, PSS) and isinstance(padding._mgf, MGF1):
-            # SHA1 is permissible in MGF1 in FIPS even when SHA1 is blocked
-            # as signature algorithm.
-            if self._fips_enabled and isinstance(
-                padding._mgf._algorithm, hashes.SHA1
+            # FIPS 186-4 only allows salt length == digest length for PSS
+            if (
+                self._fips_enabled
+                and padding._salt_length != PSS.DIGEST_LENGTH
```

Comment:
teeecchnicallly I guess its ok to manually pass a salt length equal to teh digest length? I guess we don't care?

Reply:
Yeah if we ever do this in our tests it'll fail and we'll fix it 😄

Reply:
can you add a comment for future us?